easybuild/tools/filetools.py

```diff
@@ -322,23 +424,21 @@ def find_easyconfigs(path, ignore_dirs=None):
                 continue
 
             spec = os.path.join(dirpath, f)
-            _log.debug("Found easyconfig %s" % spec)
+            _log.debug("Found easyconfig %s", spec)
             files.append(spec)
 
         # ignore subdirs specified to be ignored by replacing items in dirnames list used by os.walk
-        dirnames[:] = [d for d in dirnames if not d in ignore_dirs]
+        dirnames[:] = [d for d in dirnames if d not in ignore_dirs]
 
     return files
 
 
-def search_file(paths, query, short=False, ignore_dirs=None, silent=False):
+def search_file(paths, query, short=False, ignore_dirs=['.git', '.svn'], silent=False):
```

Comment:
don't use a list a default value, since it's mutable and can get very weird... (appending to `ignore_dirs` in the function body results in actually modifying the default value)

also, use `build_option('ignore_dirs')` is case `ignore_dirs` is `None`, don't hardcode a default like this, since we already have a default defined for `--ignore-dirs` in `easybuild/tools/options.py`